src/Algebra/Ordered/Definitions.agda

```diff
@@ -0,0 +1,41 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of functions on ordered structures
+------------------------------------------------------------------------
+
+-- The contents of this module should be accessed via `Algebra.Ordered`.
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Relation.Binary.Core
+
+module Algebra.Ordered.Definitions
+  {a ℓ} {A : Set a}   -- The underlying set
+  (_≤_ : Rel A ℓ)     -- The underlying order
+  where
+
+open import Algebra.Core
+open import Function.Base using (flip)
+open import Data.Product using (_×_)
+
+Monotonic₁ : Op₁ A → Set _
+Monotonic₁ f = f Preserves _≤_ ⟶ _≤_
+
+Antitonic₁ : Op₁ A → Set _
+Antitonic₁ f = f Preserves (flip _≤_) ⟶ _≤_
+
+Monotonic₂ : Op₂ A → Set _
+Monotonic₂ f = f Preserves₂ _≤_ ⟶ _≤_ ⟶ _≤_
+
+MonotonicAntitonic : Op₂ A → Set _
```

Comment:
Out of interest are these standard names or something concocted?

Reply:
Well monotonic and antitonic are standard — though I'm thinking of renaming them to "monotone" and "antitone", which are more common, I think. (I don't remember why I went with the *tonic variants in the first place.) The combination of mono+anti isn't "standard" but I don't know a better naming scheme. I'm open to suggestions. What is certain is that there are binary operations (like subtraction, division, residuation) that are antitone in one parameter and monotone in the other (or vice-versa).

Reply:
Update: it seems both `monotone` and `monotonic` are used in different parts of the library. Might make sense to be consistent and pick one or the other. For now, I'm leaving things as they are in the PR.

Reply:
Hmm so both `monotone` and `monotonic` are used pretty rarely in full. We do however use `mono` and `antimono` to stand for `monotonic` and `antimonotonic` a lot.

Trying to think how best to reconcile this with a consistent naming scheme for `MonotonicAntitonic` etc. ...

Reply:
You may have noticed I changed the `monotonic` fields to `mono` to be more consistent with `cong` and other parts of the library. But I'm still not sure what to do with the extra definitions (`Monotonic` and friends). I'm also fine with dropping those entirely and use `Respects` directly, if you think it's cleaner.

Reply:
Changing the fields to `mono` :+1: Let's keep the names for now. The more complicated ones don't appear in any record field names yet, so it can all be sorted out by a bit of judicious deprecation if we need to later.

Reply:
SGTM.